packages/dd-trace/src/tagger.js

```diff
@@ -5,25 +5,44 @@ const log = require('./log')
 const ERROR_MESSAGE = constants.ERROR_MESSAGE
 const ERROR_STACK = constants.ERROR_STACK
 const ERROR_TYPE = constants.ERROR_TYPE
+const DD_EMPTY_USER_TAG = constants.DD_EMPTY_USER_TAG
 
 const otelTagMap = {
   'deployment.environment': 'env',
   'service.name': 'service',
   'service.version': 'version'
 }
 
-function add (carrier, keyValuePairs, parseOtelTags = false) {
+// don't allow these tags to be empty
+const skipEmptyString = new Set(['env', 'service', 'version'])
+
+function add (carrier, keyValuePairs, parseOtelTags = false, parseSpaceSeparatedTags = false) {
   if (!carrier || !keyValuePairs) return
 
   if (Array.isArray(keyValuePairs)) {
-    return keyValuePairs.forEach(tags => add(carrier, tags))
+    return keyValuePairs.forEach(tags => add(carrier, tags, parseOtelTags, parseSpaceSeparatedTags))
   }
   try {
     if (typeof keyValuePairs === 'string') {
-      const segments = keyValuePairs.split(',')
+      let segments
```

Comment:
The tagger is already pretty heavy, I think it would be worth having a benchmark for it if there isn't already one, so see how much overhead we're adding here. Ideally, this whole logic would be reworked to actually improve the overhead even though it does more things.

Reply:
I added a benchmark in benchmark/core.js if more is needed do let me know

Reply:
I don't think those benchmarks are actually used by any check, it would be better to add a sirun benchmark that adds a tag through the public API.